contents/javascript/Javascript-framework-VueJs.md

```diff
@@ -79,13 +87,163 @@ In this case, only the `root` component can be accessed in VueJs while the rest
 <b>Step 4:</b> Pass the message to the HTML file using double curly brackets.
 ```HTML
   <div id="root">
-    <h2>{{ message }}</h2>
+    <h2>{\{message}\{</h2>
   </div>
 ```
 
 <b>Step 5:</b> Open the brower and we will see "Hello World" being displayed:
 > <h2>Hello World</h2>
+<br>
+
+-----
+### Vue's Features
+Here are some features to get you familiar with Vue:
+1. **Mutating of data in the DOM**<br>
+    In Vue, the state of the data can be directly modified.
+
+    Let's say, there is a variable called `message` in your app. To modify `message`, you can do it as such:
+    ```js
+    this.message = 'Hello Space';
+    ```
+    When `message` is changed, the view will be re-rendered to show the new message. So you can say, DOM is "reacting" to `message`.
+
+    In situations when you want to tie data to the view, do ensure that this data already has an initial value, so that any subsequent updates to this data will enable view to update itself
+    ```js
+    data: {
+        this.message = '';
+    }
+    ```
+
+
+<br>
+
+2. **2-way binding**<br>
+    `v-model` is a [Vue directive](https://vuejs.org/v2/api/#v-model) used to bind the DOM input field to its data variable.
+
+    This effectively allows the DOM variables and data to be "in sync", regardless of which one is being updated first.
+    In other words, if you change the input value, the bound data will change, vice versa.
+
+    This would also reduce any extra step required to manually update the data.
+    ```html
+    <input type="checkbox", v-model=isChecked">
+        <label for="checked">Select</label>
+    ```
+
+    2-way binding is generally used for input form bindings such as checkboxes or drop-downs, where new data is entered by users and then updated in the view.
+
+<br>
+
+3. **1-way data flow**<br>
+    When you have components that are nested within each other, data can only be passed from the outer component to the inner component, via `props`, where `props` are just custom data shared between the components.
+
+    The outer and inner components are also called parent and child, respectively.
```

Comment:
Use italics when introducing new terms. e.g.,
>The outer and inner components are also called _parent_ and _child_, respectively.

Reply:
In any case, it looks like there is no need to introduce these terms parent and child as they are not needed for the explanation.